src/test/java/com/debatetimer/client/oauth/OAuthClientTest.java

```diff
@@ -0,0 +1,102 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.client.RestClientTest;
+import org.springframework.http.HttpMethod;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.test.web.client.ResponseCreator;
+import org.springframework.test.web.client.response.MockRestResponseCreators;
+import org.springframework.web.client.RestClient;
+
+@RestClientTest(OAuthClient.class)
+class OAuthClientTest {
+
+    @Autowired
+    private MockRestServiceServer mockServer;
+
+    @Autowired
+    private RestClient.Builder restClientBuilder;
+
+    @Autowired
+    private OAuthProperties oAuthProperties;
+
+    private OAuthClient oAuthClient;
+
+    @BeforeEach
+    void setUp() {
+        this.mockServer = MockRestServiceServer.bindTo(restClientBuilder).build();
+        this.oAuthClient = new OAuthClient(restClientBuilder, oAuthProperties);
+    }
+
+
+    @Nested
+    class RequestTokenExceptionHandling {
+
+        @Test
+        void 잘못된_인증코드로로_토큰_발급요청_시_클라이언트_에러를_반환한다() {
+            MemberCreateRequest memberCreateRequest = new MemberCreateRequest("invalid_code", "redirect_uri");
+            String invalidCodeResponse = """
+                    {
+                      "ErrorCode": "invalid_request",
+                      "Error": "Invalid Authorization Code"
+                    }
+                    """;
+            setMockserver(MockRestResponseCreators.withBadRequest().body(invalidCodeResponse));
+
+            assertThatThrownBy(() -> oAuthClient.requestToken(memberCreateRequest))
+                    .isInstanceOf(DTClientErrorException.class)
+                    .hasMessage(ClientErrorCode.INVALID_OAUTH_REQUEST.getMessage());
+        }
+
+        @Test
+        void 잘못된_리다이렉트_uri로_토큰_발급요청_시_클라이언트_에러를_반환한다() {
+            MemberCreateRequest memberCreateRequest = new MemberCreateRequest("code", "invalid_uri");
+            String invalidRedirectUrlResponse = """
+                    {
+                      "ErrorCode": "invalid_request",
+                      "Error": "Invalid redirect_uri : invalid_uri"
+                    }
+                    """;
+            setMockserver(MockRestResponseCreators.withBadRequest().body(invalidRedirectUrlResponse));
+
+            assertThatThrownBy(() -> oAuthClient.requestToken(memberCreateRequest))
+                    .isInstanceOf(DTClientErrorException.class)
+                    .hasMessage(ClientErrorCode.INVALID_OAUTH_REQUEST.getMessage());
+        }
+
+        @Test
+        void 그외_에러는_서버에러를_반환한다() {
+            MemberCreateRequest memberCreateRequest = new MemberCreateRequest("code", "redirect_url");
+            String otherErrorResponse = """
+                    {
+                      "ErrorCode": "invalid_request",
+                      "Error": "Unsupported grant type : client_credentials_invalid"
+                    }
```

Comment:
1. (제안) `@ParameterizedTest`해서 조금 더 다양한 데이터로 테스트를 해도 좋을 것 같아요.
2. (제안) 응답 값이 길지 않다면, 한 줄로 작성해도 좋을 것 같아요. 아니면 따로 static final 로 포멧만 분리해도 될 것 같구요. 가독성 등 여러가지를 고려해본다면 콜리가 작성한 코드도 나쁠 것 같지는 않네요.
```suggestion
            MemberCreateRequest memberCreateRequest = new MemberCreateRequest("code", "redirect_url");
            String otherErrorResponse = "{\"ErrorCode\": \"invalid_request\", \"Error\": \"%s\"}".formatted(errorMessage);
```

Reply:
✅  둘다 반영완료했습니다!